src/components/Drawer.js

```diff
@@ -0,0 +1,149 @@
+const React = require('react');
+const Velocity = require('velocity-animate');
+
+const StyleConstants = require('../constants/Style');
+
+const Icon = require('../components/Icon');
+
+const Drawer = React.createClass({
+  propTypes: {
+    duration: React.PropTypes.number,
+    easing: React.PropTypes.array,
+    navConfig: React.PropTypes.shape({
+      label: React.PropTypes.string.isRequired,
+      onNextClick: React.PropTypes.func.isRequired,
+      onPreviousClick: React.PropTypes.func.isRequired
+    }),
+    onClose: React.PropTypes.func.isRequired
+  },
+
+  getDefaultProps () {
+    return {
+      duration: 500,
+      easing: [0.28, 0.14, 0.34, 1.04]
+    };
+  },
+
+  componentDidMount () {
+    this._animateComponent({ left: '20%' });
+    this._animateBackArrow();
+    if (this.props.navConfig) {
+      this._animateNav();
+    }
+  },
+
+  componentWillUnmount () {
+    console.log('component unmounting');
+  },
+
+  _renderNav () {
+    const styles = this.styles();
+
+    return this.props.navConfig ?
+      <nav ref={(ref) => (this._nav = ref)} style={styles.nav}>
+        <Icon onClick={this.props.navConfig.onPreviousClick} size={25} style={styles.icons} type='caret-left'/>
+        {this.props.navConfig.label}
+        <Icon onClick={this.props.navConfig.onNextClick} size={25} style={styles.icons} type='caret-right'/>
+      </nav> : null;
+  },
+
+  _handleCloseClick () {
+    this._animateComponent({ left: '100%' })
+    .then(() => {
+      this.props.onClose();
+    });
+  },
+
+  _animateComponent (transition) {
+    const el = this._component;
+    const options = {
+      duration: this.props.duration,
+      easing: this.props.easing
+    };
+
+    return Velocity(el, transition, options);
+  },
+
+  _animateBackArrow () {
+    const el = this._backArrow;
+    const transition = { left: 25 };
+    const options = {
+      delay: this.props.duration,
+      duration: this.props.duration,
+      easing: this.props.easing
+    };
+
+    Velocity(el, transition, options);
+  },
+
+  _animateNav () {
+    const el = this._nav;
+    const transition = { top: '50%' };
+    const options = {
+      delay: this.props.duration,
+      duration: this.props.duration,
+      easing: this.props.easing
+    };
+
+    Velocity(el, transition, options);
+  },
+
+  render () {
+    const styles = this.styles();
+
+    return (
+      <div ref={(ref) => (this._component = ref)} style={styles.component}>
+        <header style={styles.header}>
+          <span ref={(ref) => (this._backArrow = ref)} style={styles.backArrow}>
```

Comment:
@bsbeeks I didn't remove the span here because targeting the `Icon` element directly with a ref returned a Radium element that conflicted with Velocity's animation. I could have used the Radium element to get the dom node, but the span seemed a cleaner solution.